test/datalog_tests/rules.fail.ast.expected

```diff
@@ -6,648 +6,4 @@ expecting letter or digit, "_", ".", "(" or "["
   ./test/datalog_tests/rules.fail.dl:5:1-7:1
   ./test/datalog_tests/rules.fail.dl:7:1-9:1
 
-(typedef R1 = R1{a1: string}
```

Comment:
this is a big change... can't figure out why

Reply:
This is something I missed a long time ago. This was meant to test that ddlog rejects non-linear rules. However, we now allow such rules. So I moved the test from `simple.fail.dl` to `simple.dl`, since it does not fail anymore.